tests/BenchmarkDotNet.Tests/Validators/CompilationValidatorTests.cs

```diff
@@ -28,12 +29,25 @@ public void BenchmarkedMethodNameMustNotContainWhitespaces()
                         null)
                 }, new ManualConfig());
 
-            var errors = CompilationValidator.Default.Validate(parameters);
+            var errors = CompilationValidator.Default.Validate(parameters).Select(e => e.Message);
 
-            Assert.Equal("Benchmarked method `Has Some Whitespaces` contains illegal character(s). Please use `[<Benchmark(Description = \"Custom name\")>]` to set custom display name.", 
-                errors.Single().Message);
+            Assert.Contains(errors,
+                s => s.Equals(
+                    "Benchmarked method `Has Some Whitespaces` contains illegal character(s). Please use `[<Benchmark(Description = \"Custom name\")>]` to set custom display name."));
         }
 
+        [Theory]
+        [InlineData(typeof(BenchmarkClassWithStaticMethod), true)]
+        [InlineData(typeof(BenchmarkClass<PublicClass>), false)]
+        public void Benchmark_Class_Methods_Must_Be_Non_Static(Type type, bool hasErrors)
+        {
+            // Act
+            var validationErrors = CompilationValidator.Default.Validate(BenchmarkConverter.TypeToBenchmarks(type))
+                                                               .ToList();
+            
+            // Assert
+            Assert.Equal(validationErrors.Any(), hasErrors);
```

Comment:
Please swap the arguments.
The first argument is the expected value; the second argument is the actual value.